ee/clickhouse/queries/trends/total_volume.py

```diff
@@ -38,48 +33,156 @@ def _total_volume_query(self, entity: Entity, filter: Filter, team_id: int) -> T
-                    parsed_date_from=trend_event_query.parsed_date_from,
-                    **trend_event_query.active_user_params
-                )
-            else:
-                content_sql = VOLUME_SQL.format(event_query=event_query, **content_sql_params)
-
-            null_sql = NULL_SQL.format(trunc_func=trunc_func, interval_func=interval_func)
-            params["date_from"] = format_ch_timestamp(filter.date_from or get_earliest_timestamp(team_id), filter)
-            params["date_to"] = format_ch_timestamp(filter.date_to, filter)
-            params["interval"] = filter.interval
-            final_query = AGGREGATE_SQL.format(null_sql=null_sql, content_sql=content_sql)
-            return final_query, params, self._parse_total_volume_result(filter)
+        params["date_from"] = format_ch_timestamp(filter.date_from or get_earliest_timestamp(team_id), filter)
+        params["date_to"] = format_ch_timestamp(filter.date_to, filter)
+        params["interval"] = filter.interval
+        return final_query, params, self._parse_total_volume_result(filter)
 
     def _parse_total_volume_result(self, filter: Filter) -> Callable:
         def _parse(result: List) -> List:
-            parsed_results = []
-            for _, stats in enumerate(result):
-                parsed_result = parse_response(stats, filter)
-                parsed_results.append(parsed_result)
+            # NOTE: there's some implicit coupling here:
+            #
+            #        1. the ordering of the fields we get back, (date, value)
+            #        2. that the last row is always the `WITH TOTALS` result
+            #
+            #  TODO: remove coupling
+
+            #  The last row relates to the `WITH TOTALS` from the clickhouse query
+            values_without_totals, (_, aggregated_value) = result[:-1], result[-1]
+
+            # Separate dates and values, as desired for the REST API response
+            dates, values = zip(*sorted(values_without_totals))
+
+            parsed_result = parse_response([dates, values], filter)
 
-            return parsed_results
+            # NOTE: to maintain backwards compatability here from previous behaviour, we exclude the
+            #        `aggregated_value` for some filter.display.
+            # TODO: verify if updating to always including `aggregated_value` would break anything. It
+            #       would be simpler to not have to have this special casing in here.
+            if filter.display in TRENDS_DISPLAY_BY_VALUE:
+                parsed_result["aggregated_value"] = aggregated_value
+
+            return [parsed_result]
 
         return _parse
+
+
+ZERO_FILL_TEMPLATE = """
+SELECT 
+    interval.start AS interval_start,
+
+    -- For where we don't have an aggregate value, we zero fill
+    COALESCE(aggregate.value, 0) AS value
+
+FROM (
+    -- Creates zero values for all date axis ticks for the given date_from, date_to range
```

Comment:
This is gonna cause issues.

system.query_log automatically strips newlines meaning that this query cannot be copied from there making debugging and future optimizations really annoying.

It can also lead to instances where the SQL parsing gives up due to the SQL query growing too large.

What are these comments trying to achieve? Do they belong within the sql itself?

Reply:
Yes that’s a shame.

comments here are meant to clarify the purpose of sections of the query.

I could break up the query as separate vars and use python comments? I think it will lose some clarity though.

I could also just strip out these comments at runtime?

Reply:
Look into stripping comments @hazzadous / fstrings?